docs/notebooks/Umbrella_Integration_PySAGES_HOOMD.md

```diff
@@ -0,0 +1,281 @@
+!pip install gsd
+import gsd
+import gsd.hoomd
+import numpy as np
+
+class System:
+    def __init__(self):
+        self.L = 5
+        self.N = 200
+
+def post_process_pos(snapshot):
+    box_size = snapshot.configuration.box[:3]
+    snapshot.particles.image = np.rint(snapshot.particles.position / box_size)
+    snapshot.particles.position -= snapshot.particles.image * box_size
+    return snapshot
+
+def get_snap(system):
+    L = system.L
+    snapshot = gsd.hoomd.Snapshot()
+    snapshot.configuration.box = [L, L, L, 0, 0, 0]
+
+    snapshot.particles.N = N = system.N
+
+    snapshot.particles.types = ["A", "B"]
+    snapshot.particles.position = np.zeros((N, 3))
+    snapshot.particles.velocity = np.random.standard_normal((N, 3))
+    snapshot.particles.image = np.zeros((N, 3), dtype=int)
+    snapshot.particles.typeid = np.zeros(N, dtype=int)
+    snapshot.particles.typeid += 1
+    snapshot.particles.typeid[0] = 0
+
+    rng = np.random.default_rng()
+    for particle in range(N):
+        snapshot.particles.position[particle, 0] = rng.random() * L - L / 2
+        snapshot.particles.position[particle, 1] = rng.random() * L - L / 2
+        snapshot.particles.position[particle, 2] = rng.random() * L - L / 2
+
+    snapshot.particles.position[0, 0] = -np.pi
+    snapshot.particles.position[0, 1] = -np.pi
+    snapshot.particles.position[0, 1] = -np.pi
+
+    return snapshot
+
+system = System()
+snap = get_snap(system)
+snap = post_process_pos(snap)
+snap.particles.validate()
+with gsd.hoomd.open("start.gsd", "wb") as f:
+    f.append(snap)
+
+```
+
+<!-- #region id="AgFXHafmVUAi" -->
+For this simulation, we are using the PySAGES method `UmbrellaIntegration` so we start with importing this.
+
+In the next step, we write a function, that generates the simulation context. We need to make sure, that the context can depend on the replica of the simulation along the path. PySAGES sets variable `replica_num` in the keyword arguments of the function.
+We also set some general parameters for all replicas.
+
+In contrast to the single harmonic bias simulation, the simulation now contains an external potential `hoomd.external.periodic` which changes the expected density of particles. See hoomd-blue's [documentation](https://hoomd-blue.readthedocs.io/en/stable/module-md-external.html#hoomd.md.external.periodic) for details on the potential. For this example, the potential generates the free-energy landscape we are exploring.
+<!-- #endregion -->
+
+```python colab={"base_uri": "https://localhost:8080/"} id="tG6JhN7SNpSj" outputId="979d5793-f459-4202-ac4e-74f2aaabc1f3"
+import hoomd
+import hoomd.md
+import hoomd.dlext
+
+import pysages
+from pysages.collective_variables import Component
+from pysages.methods import UmbrellaIntegration
+```
+
+```python id="RsZhjfm2U5ps"
+param1 = {"A": 0.5, "w": 0.2, "p": 2}
+
+def generate_context(**kwargs):
```

Comment:
We could think of including a one-liner docstring in functions such as this. This applies also to other notebooks.